src/jobs/microk8s_install.yml

```diff
@@ -124,24 +130,88 @@ steps:
+      steps:
+        - run:
+            name: Get codacy chart
+            command: |
+              set -x
+              git clone https://github.com/codacy/chart.git
+              cd << parameters.chart_dir_path >>
+              git checkout << parameters.chart_branch >>
+
+  - run:
+      name: Add namespace and docker secret
+      command: |
+        set -x
+        if [[ -z "$(echo $PATH | grep snap)" ]] ; then export PATH=$PATH:/snap/bin ; fi
+
+        sudo microk8s.kubectl create namespace codacy
+        sudo microk8s.kubectl create secret docker-registry docker-credentials \
+          --docker-username=$DOCKER_USERNAME --docker-password=$DOCKER_PASSWORD -n codacy
+
+  - run:
+      name: Run additional pre-install commands (if any)
+      command: |
+        set -x
+        if [[ -z "$(echo $PATH | grep snap)" ]] ; then export PATH=$PATH:/snap/bin ; fi
+        << parameters.extra_preinstall_commands >>
+
+  - when:
+      condition: << parameters.use_chart_source >>
+      steps:
+        - deploy:
+            name: Install Codacy to microk8s from source
+            command: |
+              set -x
+              if [[ -z "$(echo $PATH | grep snap)" ]] ; then export PATH=$PATH:/snap/bin ; fi
+
+              cd << parameters.chart_dir_path >>
+              sudo helm dep up codacy/
+              sudo helm upgrade --install codacy codacy/ --namespace codacy -f ../microk8s-values.yaml
+
+  - unless:
+      condition: << parameters.use_chart_source >>
+      steps:
+        - deploy:
+            name: Install Codacy to microk8s from chart repo
+            command: |
+              set -x
+              if [[ "<< parameters.codacy_version >>" == "latest" ]] ; then
+                sudo helm upgrade --install codacy << parameters.codacy_helm_repo_name >>/codacy --namespace codacy -f microk8s-values.yaml
+              else
+                sudo helm upgrade --install codacy << parameters.codacy_helm_repo_name >>/codacy --namespace codacy -f microk8s-values.yaml --version << parameters.codacy_version >>
+              fi
+
+  - when:
+      condition: << parameters.wait_ready >>
+      steps:
+        - run:
+            name: Wait for all Codacy pods to be ready
+            command: |
+              # earlier kubectl versions don't have the userful '--all' flag
+              if [[ "$(sudo microk8s.kubectl version --short --client | cut -d. -f2)" == "13" ]] ; then
+                CODACY_PODS=$(sudo microk8s.kubectl get pods -n codacy -o jsonpath='{.items[*].metadata.name}')
+                sudo microk8s.kubectl wait --for=condition=Ready pods ${CODACY_PODS} -n codacy --timeout=<< parameters.wait_ready_timeout >>s
+              else
+                sudo microk8s.kubectl wait --for=condition=Ready pods --all -n codacy --timeout=<< parameters.wait_ready_timeout >>s
+              fi
+
+              # show state before exiting
+              sudo microk8s.kubectl get pods -n codacy
+
+  - run:
```

Comment:
This is for sure useless as the user can always use a different step for doing the same.

Reply:
👍